### PR TITLE
fix: make tedge-write not reset owner/group

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

Fixed two closely related problems in tedge-write:

- When destination file existed, its owner/group was reset because when creating a temporary file for atomic write we forgot to set the same permissions as the destination file. Now we copy destination file permissions to the temporary, ensuring that after the rename permissions stay the same.
- Changed some permissions used in the tests from 644 to 666, so the tests don't fail when default umask is different.

Also added new functionality: tedge-write can now use plain uid/gid numbers to change owner instead of needing a name. This was done so we can more easily test file owner changes, but is probably a user improvement in its own right.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
- #2972

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_write/tests/main.rs [hunk 1]
```diff
@@ -31,49 +33,59 @@ fn creates_dest_file_if_doesnt_exist() {
 }
 
 #[test]
-fn sets_file_permissions_if_file_doesnt_exist() {
+fn changes_permissions_if_destination_doesnt_exist() {
     // Arrange
     let (temp_dir, source_path) = setup_source_file();
-    fs::set_permissions(&source_path, Permissions::from_mode(0o644)).unwrap();
+    chown(&source_path, Some(2137), Some(2137)).unwrap();
```
Comment (didier-wenzek) [inline crates/core/tedge_write/tests/main.rs:39]: Such a test can only be successful if run by a privileged user allowed to `chown` a file.

**Review by didier-wenzek — COMMENTED** [review]
Adding an option to use plain uid/gid numbers is a good idea. And the fix seems correct. But, you have to come with another way to test it without assuming chown privileges.

File: crates/core/tedge_write/tests/main.rs (hunk 1 above)
Comment (Bravo555) [reply crates/core/tedge_write/tests/main.rs]: Rust integration tests were a bad environment for such a test, they were moved to RobotFramework in 43f5705.

**Review by didier-wenzek — COMMENTED** [review]
Indeed replacing system tests by integration tests is more appropriate. The code being correct, I will approve this PR once rebased with the system-tests fix.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**